source/common/common/hash.h

```diff
@@ -135,11 +134,7 @@ class MurmurHash {
   static uint64_t murmurHash2(absl::string_view key, uint64_t seed = STD_HASH_SEED);
 
 private:
-  static inline uint64_t unalignedLoad(const char* p) {
-    uint64_t result;
-    safeMemcpyUnsafeSrc(&result, p);
-    return result;
-  }
+  static inline uint64_t unalignedLoad(const char* p) { return absl::little_endian::Load64(p); }
```

Comment:
can we have a unit test?

The PR description says this adds new functionality, but you are changing an existing implementation. Are we sure this is correct?

Also, is this needed for a hash?  Ordinarily I would think for a hash it would be OK to have a different answer on little-endian vs big-endian machines but maybe this is violating some property of Murmur hash?

Comments would be good.

Reply:
Thanks for reviewing @jmarantz. Yes, it should still be correct. Under the hood, `absl::little_endian::Load64` does a `memcpy` like the orignal code except that on big endian platforms it will also do a byte swap.

That's a good point. The original murmurhash code points out where byte swapping needs to be done on big endian platforms to get the same values as little endian but it also mentions that the properties of the hash are not affected by endianness. I'd be happy to stick with the original `unalignedLoad` and instead add big endian expected values to the test if that makes sense.

Reply:
@jmarantz, I pushed a new commit that makes the hash values the same on all platforms regardless of endianness and doesn't use the `absl::little_endian::Load64` function.

The existing //test/extensions/load_balancing_policies/ring_hash:ring_hash_lb_test test covers this change; it fails on big endian platforms without it. I've added comments to hopefully make it clear why byte swapping is applied.

Hope that all looks good now. I think its a bit better to apply the byte swapping rather than just add expected values in the text for big endian but happy to discuss.